### ISSUE TITLE
Handle ampersands

### ISSUE THREAD

**Issue by jeremyhansonfinger** [Shopify/retext-shopify#12]
Ampersands in the ruleset are currently ignored.

**Comment by wooorm** [comment]
@jeremyhansonfinger What’s the issue with ampersands? Maybe I can help?!

P.S. I’m recently adding [some more plugins to retext](https://github.com/wooorm/retext/blob/master/doc/plugins.md#list-of-plugins). Might be something useful in there (for example, [indefinite-article](https://github.com/wooorm/retext-indefinite-article), [repeated-words](https://github.com/wooorm/retext-repeated-words)).

**Comment by jeremyhansonfinger** [comment]
@wooorm This is probably pretty simple, but I can't figure out how to properly escape special characters when creating a rule and writing a test for it.

So far I've tried `&`, `\&`, and `%26`.

Rule:

```json
  "&": {
    "replace": "and",
    "note": "Don't use ampersands."
  },
```

Test:

```js
test('ampersand', function (t) {
    var actual;

    t.plan(2);

    actual = process('bacon & Eggs.');
    t.equal(
        actual.messages[0].ruleId,
        '&',
        '“&” violates ruleId “&”'
    );
    t.equal(
        actual.messages[0].source,
        'retext-shopify',
        'source is “retext-shopify”'
    );
});
```

**Comment by jeremyhansonfinger** [comment]
It fails with `TypeError: Cannot read property 'ruleId' of undefined`

**Comment by wooorm** [comment]
Ah, alright. So, the deal is that [`nlcst-search`](https://github.com/wooorm/nlcst-search) searches for words only, and `&` isn’t a word (it’s a symbol).

First: are you trying to ban all ampersands? Is `AT&T` wrong too? Or are loose ampersands wrong (`&`), and/or use in words like so: `&c.`?

`AT&T` and `&c.` are both words, so you should be able to search for those.

Loose ampersands, are as mentioned before, symbols, and those must be searched for using different techniques.

Something like:

```js
var visit = require('unist-util-visit');
var toString = require('nlcst-to-string');
// `file` and `tree` are given to the transformer.
visit(tree, 'SymbolNode', function (node) {
  if (toString(node) === '&') {
    file.warn('Do not use ampersands', node, 'ampersand');
  }
});
```

**Comment by jeremyhansonfinger** [comment]
Right, of course. I forgot about how the tree is constructed of punctuation elements and word elements.

**Comment by wooorm** [comment]
Yeah that's something to keep in mind!
Note that it's also possible to regex over the whole file, by coercing it to string ('file.toString'),
And checking that. Then, use [vfile-location](https://github.com/wooorm/vfile-location) to get positions from offsets.
